app/routes.py

```diff
@@ -251,3 +328,114 @@ def get_ags_log(bgs_loca_id: int = ags_log_query,
+    """
+    Export a single borehole in .ags format from AGS data held by the National Geoscience Data Centre.
+    :param bgs_loca_id: The unique identifier of the borehole to export.
+    :type bgs_loca_id: str
+    :return: A response containing a .zip file with the exported borehole data.
+    :rtype: Response
+    :raises HTTPException 500: If the borehole exporter could not be reached.
+    :raises HTTPException 404: If the specified borehole does not exist or is confidential.
+    :raises HTTPException 500: If the borehole exporter returns an error.
+    """
+
+    url = BOREHOLE_EXPORT_URL.format(bgs_loca_id=bgs_loca_id)
+
+    try:
+        response = requests.get(url, timeout=10)
+    except (Timeout, ConnectionError):
+        raise HTTPException(status_code=500,
+                            detail="The borehole exporter could not be reached.  Please try again later.")
+
+    try:
+        response.raise_for_status()
+    except HTTPError:
+        if response.status_code == 404:
+            raise HTTPException(status_code=404,
+                                detail=f"Failed to retrieve borehole {bgs_loca_id}. "
+                                "It may not exist or may be confidential")
+        else:
+            raise HTTPException(status_code=500,
+                                detail="The borehole exporter returned an error.")
+
+    headers = {'Content-Disposition': 'attachment; filename="boreholes.zip"'}
+
+    return Response(response.content, headers=headers, media_type='application/x-zip-compressed')
+
+
+@router.get("/ags_export_by_polygon/",
+            tags=["ags_export_by_polygon"],
+            summary="Export a number of boreholes in .ags format",
+            description=("Export a number of boreholes in .ags format from AGS data "
+                         "held by the National Geoscience Data Centre."),
+            include_in_schema=True,
+            response_class=Response,
+            responses=zip_ags_responses)
+def ags_export_by_polygon(polygon: str = polygon_query):
+    """
+    Export the boreholes in .ags format from AGS data held by the National Geoscience Data Centre,
+    that are bounded by the polygon. If there are more than 10 boreholes return an error
+    :param polygon: A polygon in Well Known Text.
+    :type polygon: str
+    :return: A response containing a .zip file with the exported borehole data.
+    :rtype: Response
+    :raises HTTPException 404: If there are no boreholes or more than 10 boreholes in the polygon.
+    :raises HTTPException 422: If the WEll Known Text is not a POLYGON or is invalid.
+    :raises HTTPException 500: If the borehole index could not be reached.
+    :raises HTTPException 500: If the borehole index returns an error.
+    :raises HTTPException 500: If the borehole exporter could not be reached.
+    :raises HTTPException 500: If the borehole exporter returns an error.
+    """
+
+    # Check explicitly that the WKT is a valid POLYGON
```

Comment:
This block could be replaced if the upstream API at BOREHOLE_INDEX_URL reported invalid polygons (start point and end point not matching) correctly.